include/wx/msw/ole/safearray.h

```diff
@@ -367,7 +367,7 @@ class wxSafeArray : public wxSafeArrayBase
             result = sa.ConvertToVariant(variant);
 
         if ( sa.HasArray() )
-            sa.Detach();
+            static_cast<void*>(sa.Detach()); // we do not own the array, just attached it
```

Comment:
Could/should we use `wxUnusedVar()` instead of the cast? It would seem to be more clear, although the most clear would probably be to rename that function (while keeping the current name for compatibility, of course...) to `wxIgnoreUnusedWarning()` as then the code using it would really be self-documenting.

Reply:
As I wrote elsewhere, I realized that the cast cannot be used since the warning is produced with clang and GCC.

I think for now I am going to use `wxUnusedVar()` and the renaming of the function should be left to a different PR.

Reply:
I did see that you write it, but then I've also seen that the CI builds passed and so thought you've fixed it by using static cast instead of a plain cast or something like this. If not, I don't understand why did they pass, they're supposed to use `-Werror` and fail if there are any warnings... Do you understand what's going on here or should I look at this?

Reply:
(Sorry, I missed this comment before).

I did not not know that (some?) CI builds do not allow any warnings. I get a few between the library and samples when building with GCC locally (at least when using CMake).

I am ashamed to admit I cannot reproduce the GCC warning when reverting from `wxUnusedVar` to `static_cast<void*>` anymore, I can't believe that extending the macro definition would change that (tried with both 11 and 17 C++ standards). But it seems that generally the warnings are always not the same (depends on C++ standard, e.g., the string template one from the XML library ), but perhaps I just miss them...